src/main/java/org/terasology/flexiblepathfinding/JPSImpl.java

```diff
@@ -70,21 +63,17 @@ public static void setStatsEnabled(boolean statsEnabled) {
      */
     public boolean run() throws InterruptedException {
         startMillis = System.currentTimeMillis();
-
-        Callable<Boolean> callable = () -> performSearch();
-        boolean result = false;
-        if (this.timeLimiter != null) {
-            try {
-                result = timeLimiter.callWithTimeout(callable, (long) (config.maxTime * 1000.0f), TimeUnit.MILLISECONDS);
-            } catch (Exception e) {
-                logger.warn("Timeout of {} exceeded: {}", config.maxTime, e.toString());
-            }
-        } else {
-            result = performSearch();
+        try {
+            boolean result = Boolean.TRUE.equals(Mono.fromCallable(this::performSearch)
+                    .subscribeOn(GameScheduler.parallel())
+                    .block(Duration.ofMillis((long) (config.maxTime * 1000.0f))));
```

Comment:
We want to be very careful about when we call `block`, right?

This looks weird, because if I understand correctly, this `JPSImpl#run` is called by `PathfinderSystem#processPath`, which was _already_ sent to run on `GameScheduler.parallel()` by `PathfinderSystem#requestPath`.

So this is executing on the parallel scheduler, and it adds a new thing to that scheduler, and then _blocks one of that scheduler's threads_—that sounds like a good way to get a logjam, requiring later tasks to complete before earlier ones yield.

It also looks like the old implementation _only_ did that `timeLimiter` stuff **if** `config.executor` was non-null. Can we get away with dropping that feature?

Reply:
lets do that in another PR. this is just migrating the existing code over.

Reply:
Oh, I found where it set `config.executor`: it was in PathfinderSystem.requestPath. It looks like it did have a _second_ task queue for that purpose. That means this replacement where it's using the same scheduler for both is not equivalent.

Reply:
I reverted the removal of the `TimeLimiter` usage as I don't understand why it is necessary for replacing `TaskMaster` with Reactor.
Also, I believe I'm using a new thread for the code path without the time limiter now, so the issue @skaldarnar reported below should be gone. At least when testing this in `CoreGameplay` + `WildAnimals` by spawning a deer, I didn't get any exceptions as far as I can see.